core/lib/config/src/configs/da_client/mod.rs

```diff
@@ -0,0 +1,19 @@
+use serde::Deserialize;
+
+use crate::{AvailConfig, ObjectStoreConfig};
+
+pub mod avail;
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+pub struct DAClientConfig {
+    #[serde(flatten)]
+    pub client: DAClient,
+}
+
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(tag = "client")]
+pub enum DAClient {
+    NoDA,
```

Comment:
There seem to be a confusion in logic with this variant. You can serialize `NoDA` using env config, but not with protobuf config. Simultaneously, both lack of config and `NoDA` config result in using `NoDAClientWiringLayer`.

Do we need it as an explicit option? What would be the use cases here?
If we can detect a misconfiguration based on this (e.g. no config was passed in one place, but a related config was placed elsewhere), then deserialization should probably be explicit, and `Some(NoDA)` should be treated differently from `None`. If not, probably we don't need `NoDA` variant at all.

Also, if you keep it, please leave a comment explaining how this option can be used (e.g. the application should err if there is no `DAClient` config provided, but `DADispatcher` config is present).